sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java

```diff
@@ -46,6 +46,11 @@
    */
   String PROP_LOCATION = "location";
 
+  /**
+   * A reserved property to specify a table was created with EXTERNAL.
+   */
+  String PROP_EXTERNAL = "external";
```

Comment:
Spark hides the `EXTERNAL TABLE` concept from users, that's why we forbid `EXTERNAL` in the native CREATE TABLE syntax. I don't think we should expose the `EXTERNAL` to catalog plugins.

Reply:
Because `LOCATION` supersedes `EXTERNAL` and all external tables require location, right? If my understanding is correct, I agree `PROP_EXTERNAL` is redundant and `PROP_LOCATION` can be used instead.

Reply:
It is up to source providers how to handle this. If the parser accepts `EXTERNAL` then the fact that it was added should be passed to sources.

Reply:
The parse doesn't accept `EXTERNAL` except for hive serde tables. I don't agree that because Hive accepts it then all Spark data sources should accept it.

Why not forbid EXTERNAL for hive serde tables? It can be an option as well.

Reply:
What's the goal here? Would we want to pass all the configs to source provider "transparently"? If then yes it makes sense, but if not is there any case which LOCATION doesn't supersede EXTERNAL?

Reply:
The v2 API is not the right place to make this decision. If we support an option in the parser, the source should know whether it was used, or else you are making assumptions about how the source will operate.